subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/report/AbstractTablePageGenerator.java

```diff
@@ -262,4 +266,12 @@ private void renderVersionHeader(String version) {
             }
             // @formatter:on
     }
+
+    private String determineBaseline() {
+        return executionDataProvider.getReportScenarios().stream()
+            .filter(scenario -> !scenario.getCrossBuild())
+            .findFirst()
+            .map(scenario -> scenario.getHistoryExecutions().isEmpty() ? "" : scenario.getHistoryExecutions().get(0).getBaseVersion().getName())
+            .orElse("");
```

Comment:
Why use an empty string here instead of `null`? Or even better, return the optional here.